api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java

```diff
@@ -348,10 +349,14 @@ default void retrieveTransactionHistory(
      *
      * @param player       the {@link UUID} of the potential member
      * @param subscription the {@link EconomySubscriber} accepting whether the user is a member
-     * @author lokka30
+     * @author lokka30, MrIvanPlays
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    void isMember(@NotNull UUID player, @NotNull EconomySubscriber<Boolean> subscription);
+    default void isMember(@NotNull UUID player, @NotNull EconomySubscriber<Boolean> subscription) {
+        Objects.requireNonNull(player, "player");
+        Objects.requireNonNull(subscription, "subscription");
+        hasPermission(player, subscription, AccountPermission.BALANCE); // permission doesn't matter
```

Comment:
A player is an account member as long as they have at least one permission. By assuming that any member will have at least `BALANCE` we don't follow that description as someone may have the permission to deposit but not consult. If that makes sense or not is left for the provider to define

Reply:
this means that `Account` won't have a default implementation for `isMember`.

Reply:
It could have by checking each permission individually and when you find one that the player has succeeded with true, if no permission is found succeeded with false

Reply:
> It could have by checking each permission individually and when you find one that the player has succeeded with true, if no permission is found succeeded with false

This kills the spirit of SQL queries if anyone's going to use mysql.

Reply:
You can also use `retrievePermissions` and see if at least one is set to true

Reply:
> You can also use `retrievePermissions` and see if at least one is set to true

why should we load all the permissions when we can just do a simple index check in mysql? the counterpoint of this would be that currently the permissions are only 4, but let it stay as is.